File: Strivers/src/basicrecursion/nto1/Nto1.java
package basicrecursion.nto1;

import java.util.Scanner;

public class Nto1 {
    public static void main(String[] args) {
        System.out.print("Enter the number : ");
        Scanner sc = new Scanner(System.in);
        int number = sc.nextInt();
        printNto1(number);
    }

    private static void printNto1(int n) {
        if (n <= 0) ;
        else {
            printNto1(n - 1);
            System.out.print(n + " ");
        }
    }
}

File: Strivers/src/basicrecursion/nto1/Nto1.kt
package basicrecursion.nto1

import java.util.Scanner

fun main(args: Array<String>){
    val sc = Scanner(System.`in`)
    print("Enter the number : ")
    val number = sc.nextInt()
    printNto1(number)
}

fun printNto1(n: Int): Unit {
    if (n <= 0) return
    else {
        printNto1(n-1)
        print("$n ")
    }
}